### DISCUSSION TITLE
2 Asset Backtest (Basis, pairs, 2 exchange arb)

### DISCUSSION THREAD

**Discussion by degloff** [nkaz001/hftbacktest#35] [Ideas]
Currently there is SingleAssetHftBacktest. I looked into the possibilities how to extend hftbacktest to also allow 2 assets for backtesting, where the data is coming from 2 exchanges with a latency between the data. A quick approach would be to create more data types (another set of market data for the 2nd exchange) and extend SingleAssetHftBacktest. It is not so clear though how the order matching should be done.

**Comment by nkaz001** [comment]
It's not as easy as it appears to implement generalized code due to the limitations of Numba. So, I'm rewriting it in Rust.

But, you can still achieve this by modifying `SingleAssetHftBacktest` yourself. It should contain two local processors and two exchange processors. And you'll need to implement `goto` method, which runs each processor in time order.

**Comment by nkaz001** [comment]
It's still at the very early stage of development, I updated Rust implementation that supports multiple assets. https://github.com/nkaz001/hftbacktest/tree/experimental_rust
